projects/Mallard/src/helpers/settings/resolvers.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * This exposes Apollo resolvers for all setting as well as query helpers for
+ * some of the GDPR settings.
+ */
+import { getSetting, GdprBuckets, GDPRBucketKeys } from 'src/helpers/settings'
+
+export const ALL_GDPR_SETTING_NAMES = Object.keys(GdprBuckets).reduce<string[]>(
+    (names, bucketName) => {
+        return names.concat([bucketName], GdprBuckets[
```

Comment:
As we've discussed offline, we store setting values both for GDPR bucket names, and for specific GDPR setting names. So that's indeed the goal here to contact both keys and values. I can definitely add a comment to explain that.

The `[]` around `bucketName` is optional but type-wise it is slightly more correct. Happy to change if Typescript doesn't complain.

Reply:
Ah cool, sorry yeah forgot about what that data structure represented - all is good 👌